pkg/icingadb/db.go

```diff
@@ -495,6 +553,21 @@ func (db *DB) Delete(ctx context.Context, entityType contracts.Entity, ids []int
 	return db.DeleteStreamed(ctx, entityType, idsCh)
 }
 
+var placeholder = regexp.MustCompile(`\?`)
+
+// PostProcessPlaceholders returns query with placeholders (?) suitable for db.
+func (db *DB) PostProcessPlaceholders(query string) string {
+	if db.DriverName() == "icingadb-pgsql" {
+		var i uint64
+		return placeholder.ReplaceAllStringFunc(query, func(string) string {
+			i++
+			return "$" + strconv.FormatUint(i, 10)
+		})
+	}
+
+	return query
+}
```

Comment:
Is there no way to have `database/sql` or `sqlx` handle this for us?

Reply:
Should there be one? This one converts one kind of placeholders to another, whereas `database/sql` is completely agnostic, the drivers only care for their own kind of placeholders and `sqlx` is about specific query args which are not involved here.

Reply:
There is https://pkg.go.dev/github.com/jmoiron/sqlx#DB.Rebind which does pretty much the same thing. There's also https://pkg.go.dev/github.com/jmoiron/sqlx#BindDriver, maybe we should also register the `icingadb-mysql`/`icingadb-pgsql` drivers there (might even be needed to use `DB.Rebind()`)?